tests/torchtune/datasets/test_chat_dataset.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from unittest import mock
+
+import pytest
+from torchtune.data import Message
+from torchtune.data._common import CROSS_ENTROPY_IGNORE_IDX
+
+from torchtune.datasets import ChatDataset
+
+
+class DummyTokenizer:
+    def encode(self, text, **kwargs):
+        words = text.split()
+        return [len(word) for word in words]
+
+    @property
+    def eos_id(self):
+        return -1
+
+
+class DummyTemplate:
+    def __init__(self):
+        self.template = {
+            "system": "System:\n{system}\nUser:\n{user}\nAssistant:\n",
+            "no_system": "User:\n{user}\nAssistant:\n",
+        }
+
+    def format(self, sample, column_map=None):
+        if "system" in sample:
+            return self.template["system"].format(**sample)
+        else:
+            return self.template["no_system"].format(**sample)
+
+
+def _are_messages_equal(messages_a, messages_b):
+    for ma, mb in zip(messages_a, messages_b):
+        if ma.role != mb.role:
+            return False
+        if ma.content != mb.content:
+            return False
+    return True
+
+
+class TestChatDataset:
+    @pytest.fixture
+    def template(self):
+        return DummyTemplate()
+
+    @pytest.fixture
+    def dialogue(self):
+        return [
+            {
+                "dialogue": [
+                    Message(role="system", content="You are an AI assistant."),
+                    Message(role="user", content="What is the meaning of life?"),
+                    Message(role="assistant", content="The meaning of life is 42."),
+                    Message(role="user", content="That's ridiculous."),
+                    Message(role="assistant", content="I agree."),
+                ],
+            },
```

Comment:
I'm a bit confused.. why isn't dialogue just a list of messages? Why the extra wrapping in dict with just "dialogue" key?

Reply:
made it this way to represent real data, which would be a row in a data table with various fields, and the conversations field points to the actual list of messages.